mautrix/bridge/user.py

```diff
@@ -162,9 +166,34 @@ async def push_bridge_state(
         if state.should_deduplicate(self._prev_bridge_status):
             return
         self._prev_bridge_status = state
-        await state.send(
-            self.bridge.config["homeserver.status_endpoint"], self.az.as_token, self.log
-        )
+        self._bridge_state_queue.append(state)
+        if not self._bridge_state_loop:
+            self.log.debug(f"Starting bridge state loop")
+            self._bridge_state_loop = asyncio.create_task(self._start_bridge_state_send_loop())
+        else:
+            self.log.debug(f"Queued bridge state to send later: {state.state_event}")
+
+    async def _start_bridge_state_send_loop(self):
+        while self._bridge_state_queue:
+            state = self._bridge_state_queue[0]
+            success = await state.send(
+                self.bridge.config["homeserver.status_endpoint"], self.az.as_token, self.log
+            )
+            if success:
+                self._bridge_state_queue.popleft()
+            else:
+                if state._num_send_attempts <= 10:
+                    retry_seconds = state._num_send_attempts ** 2
+                    self.log.warn(
+                        f"Error sending bridge state {state.state_event}, attempts: {state._num_send_attempts}, waiting {retry_seconds} seconds for retry"
+                    )
+                    await asyncio.sleep(retry_seconds)
+                else:
+                    self.log.error(
+                        f"Failed to send bridge state {state.state_event} after {state._num_send_attempts} tries, giving up"
+                    )
+                    self._bridge_state_queue.pop(0)
```

Comment:
```suggestion
                    self._bridge_state_queue.popleft()
```

Reply:
uggggg how did i miss that one? pyright was even mad about it :(